bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Bookie.java

```diff
@@ -441,77 +335,122 @@ private void checkIfDirsOnSameDiskPartition(List<File> dirs) throws DiskPartitio
         }
     }
 
+    static Versioned<Cookie> readAndVerifyCookieFromRegistrationManager(Cookie masterCookie,
+                                                                        ServerConfiguration conf,
+                                                                        RegistrationManager rm)
+            throws BookieException {
+        // we need to loop through all possible bookie identifiers to ensure it is treated as a new environment
+        // just because of bad configuration
+        List<BookieSocketAddress> addresses = Lists.newArrayListWithExpectedSize(3);
+        try {
+            // ip address
+            addresses.add(getBookieAddress(
+                new ServerConfiguration(conf).setUseHostNameAsBookieID(false).setAdvertisedAddress(null)));
+            // host name
+            addresses.add(getBookieAddress(
+                new ServerConfiguration(conf).setUseHostNameAsBookieID(true).setAdvertisedAddress(null)));
+            // advertised address
+            if (null != conf.getAdvertisedAddress()) {
+                addresses.add(getBookieAddress(conf));
```

Comment:
I wonder if the advertised address should be really "known" (does not throw UnknownHostException).
it is an address used by other machines to refer to "this" machine.
Is this a behaviour change ?

Reply:
I don't understand the comment here. can you explain more?

Reply:
in case of UnkownHostException while dealing with the advertised address we are failing the boot.
it is possible that this name is not known/resolvable to the bookie because it is a name from another "namespace"

Reply:
discussed on slack, there is no change from prev version so it is ok for me